exasol/ai/mcp/server/mcp_server.py

```diff
@@ -0,0 +1,366 @@
+        Builds a metadata query.
+
+        Args:
+            meta_name:
+                Must be one of "SCHEMA", "TABLE", "VIEW", "FUNCTION", or "SCRIPT".
+            conf:
+                Metadata type settings, which is a part of the server configuration.
+            schema_name:
+                The schema name provided in the call to the tool. In case it's empty
+                the current schema of the pyexasol connection may be used. If the
+                is no current schema the query will return objects from all schemas.
+            predicates:
+                Any additional predicates to be used in the WHERE clause.
+        """
+        predicates = [conf.select_predicate, *predicates]
+        if meta_name != "SCHEMA":
+            schema_name = schema_name or self.connection.current_schema()
+        if schema_name:
+            predicates.append(f"{meta_name}_SCHEMA = '{schema_name}'")
+        return dedent(
+            f"""
+            SELECT {meta_name}_NAME AS "{conf.name_column}", {meta_name}_COMMENT AS "{conf.comment_column}"
+            FROM SYS.EXA_ALL_{meta_name}S
+            {_where_clause(*predicates)}
+        """
+        )
+
+    def _execute_query(
+        self, tool_name: str, query: str, use_snapshot: bool = True
+    ) -> TextContent:
+        """
+        Executes the specified query and returns the result data in the form of json
+        wrapped in a TextContent. For example
+        TextContent(
+            type='text',
+            text='[{"name": "MY_TABLE1", "comment": "first table"}, {"name": "MY_TABLE2", "comment": "second table"}]'
+        )
+        If an error occurs, returns
+        TextContent(
+            type='text',
+            text='{"error": "<error stack>"}'
+        )
+
+        By default, executes a lock-free request using the pyexasol
+        `meta.execute_snapshot` method. This is a recommended way of running a metadata
+        query. For a normal query the `use_snapshot` should be set to `False`.
+        """
+        try:
+            if use_snapshot:
+                result = self.connection.meta.execute_snapshot(query=query).fetchall()
+            else:
+                result = self.connection.execute(query=query).fetchall()
+            result_json = json.dumps(result)
+            return TextContent(type="text", text=result_json)
+        except Exception as e:  # pylint: disable=broad-exception-caught
+            return _report_error(tool_name, str(e))
+
+    def list_schemas(self) -> TextContent:
+        tool_name = self.list_functions.__name__
+        conf = self.config.schemas
+        query = self._build_meta_query("SCHEMA", conf, "")
+        return self._execute_query(tool_name, query)
+
+    def list_tables(
+        self,
+        schema_name: Annotated[
+            str, Field(description="name of the database schema", default="")
+        ],
+    ) -> TextContent:
+        tool_name = self.list_tables.__name__
+        t_conf = self.config.tables
```

Comment:
Name is not very expressive:
```suggestion
        table_conf = self.config.tables
```